src/aliceVision/hdr/LaguerreBACalibration.cpp

```diff
@@ -19,291 +19,284 @@
 
-    ALICEVISION_LOG_DEBUG("Create exposure list");
-    std::map<std::pair<int, float>, double> exposures;
-    for(int i = 0; i < cameraExposures.size(); ++i)
-    {
-        const std::vector<float>& camExp = cameraExposures[i];
-        for (int j = 0; j < camExp.size(); ++j)
-        {
-            const auto& exp = camExp[j];
 
-            ALICEVISION_LOG_TRACE(" * " << imagePathsGroups[i][j] << ": " << exp);
+    std::vector<std::vector<double>> exposuresRatios;
 
-            // TODO: camId
-            exposures[std::make_pair(0, exp)] = exp;
+    for (std::vector<float> & group : cameraExposures) {
+        std::vector<double> dest;
+
+        for (int index = 0; index < group.size() - 1; index++) {
+            
+            double exposure_first = group[index];
+            double exposure_second = group[index + 1];
+            dest.push_back(exposure_second / exposure_first);
         }
+        
+        exposuresRatios.push_back(dest);
     }
-    std::array<double, 3> laguerreParam = { 0.0, 0.0, 0.0 };
-    std::array<double, 3> relativeWB = { 1.0, 1.0, 1.0 };
 
-    ALICEVISION_LOG_DEBUG("Create BA problem");
+    std::array<double, 3> laguerreParam = {0.0, 0.0, 0.0};
+
     ceres::Problem problem;
+    ceres::LossFunction* lossFunction = nullptr;
 
-    // Should we expose the LOSS function parameter?
-    ceres::LossFunction* lossFunction = new ceres::HuberLoss(Square(0.12)); // 0.12 ~= 30/255
-
-    // In the Bundle Adjustment each image is optimized relativelty to the next one. The images are ordered by exposures.
-    // But the exposures need to be different between 2 consecutive images to get constraints.
-    // So if multiple images have been taken with the same parameters, we search for the next closest exposure (larger or smaller).
-    // For example a dataset of 6 images with 4 different exposures:
-    //       0        1        2        3        4        5
-    //     1/800    1/800 -- 1/400 -- 1/200    1/200 -- 1/100
-    //       \_________________/        \________________/
-    // Without duplicates the relative indexes would be: [1, 2, 3, 4, 5, 4]
-    // In this example with 2 duplicates, the relative indexes are: [2, 2, 3, 5, 5, 3]
-    std::vector<std::vector<int>> closestRelativeExpIndex;
+    
+    // Convert selected samples into residual blocks
+    for (int groupId = 0; groupId < samples.size(); ++groupId)
     {
-        closestRelativeExpIndex.resize(cameraExposures.size());
-        for(int i = 0; i < cameraExposures.size(); ++i)
-        {
-            const std::vector<float>& camExp = cameraExposures[i];
-            std::vector<int>& camRelativeExpIndex = closestRelativeExpIndex[i];
-            camRelativeExpIndex.resize(camExp.size(), -1);
-            for(int j = 0; j < camExp.size(); ++j)
+        std::vector<ImageSamples> & group = samples[groupId];
+
+        for (int imageId = 0; imageId < group.size() - 1; imageId++) {
+
+            const ImageSamples & samples = group[imageId];
```

Comment:
a variable `samples` is already declared at line 181.
It is better to rename it and even better use the range loops where possible